### PR TITLE
chore: add resource_pools intg test

### PR DESCRIPTION
**Pull request by carolinaecalderon**
## Ticket

RM-201

## Description
Add intg tests for resource_pool.go
Leave a couple things as TODO for Bradley's refactor PR.

## Test Plan
See attached test.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/rm/kubernetesrm/resource_pool_intg_test.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+	rp, jobID := testResourcePoolWithJob(t, defaultSlots)
+
+	allocID := newTestPod(t, rp, jobID)
+	require.True(t, rp.reschedule)
+	require.False(t, rp.reqList.IsScheduled(allocID))
+
+	rp.Schedule()
+
+	require.False(t, rp.reschedule)
+	go require.True(t, rp.reqList.IsScheduled(allocID))
```
Comment (davidfluck-hpe) [inline master/internal/rm/kubernetesrm/resource_pool_intg_test.go:173]: Why is this one a goroutine and not the others? (I apologize, I'm not familiar with this part of the code. I dug into it a bit, but couldn't find anything.)

Comment (stoksc) [reply master/internal/rm/kubernetesrm/resource_pool_intg_test.go:173]: I think @carolinaecalderon was trying to fix the race condition. This (surprisingly) tricks the race detector but, of course, doesn't fix it. I think she's working on something else.

Comment (carolinaecalderon) [reply master/internal/rm/kubernetesrm/resource_pool_intg_test.go]: yeah, outdated -- I removed the test that was flaky in CI. I created a new ticket (https://hpe-aiatscale.atlassian.net/browse/RM-291) where I'll address this later.

**Review by dannysauer — APPROVED** [review]
LGTM

**Review by davidfluck-hpe — APPROVED** [review]

**Review by stoksc — APPROVED** [review]

**Pull request merged**